pkg/registry/common/memory/ns_server.go

```diff
@@ -59,6 +61,11 @@ func (s *memoryNSServer) Register(ctx context.Context, ns *registry.NetworkServi
 		return nil, err
 	}
 
+	loaded, ok := s.networkServices.LoadOrStore(r.Name, r.Clone())
+	if ok && proto.Equal(ns, loaded) {
+		return nil, errors.New("network service with this name already exist")
+	}
```

Comment:
I thought memory registry is not smart and it just replaces entries. 
I also don't quite follow why we change logic only for networkservices and don't touch networkserviceendpoints..